utils/test/compare.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package test
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"os"
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+func CompareStrings(actual, expected []string) (unexpected, missing []string) {
+	for _, res := range actual {
+		if !containsString(expected, res) {
+			unexpected = append(unexpected, res)
+		}
+	}
+	for _, expRes := range expected {
+		if !containsString(actual, expRes) {
+			missing = append(missing, expRes)
+		}
+	}
+	return
+}
+
+func CompareResources(actual, expected []func() resource.Resource) (unexpected, missing []string) {
+	for _, res := range actual {
+		if !containsResource(expected, res) {
+			unexpected = append(unexpected, getPackageAndNameOfResource(res))
+		}
+	}
+	for _, expRes := range expected {
+		if !containsResource(actual, expRes) {
+			missing = append(missing, getPackageAndNameOfResource(expRes))
+		}
+	}
+	return
+}
+
+func CompareDataSources(actual, expected []func() datasource.DataSource) (unexpected, missing []string) {
```

Comment:
Should we also generilize the implementation like this: 
```
func CompareResources(actual, expected []func() resource.Resource) (unexpected, missing []string) {
	actualNames := ToStringSliceFunc(actual, getPackageAndNameOfResource)
	expectedNames := ToStringSliceFunc(expected, getPackageAndNameOfResource)
	return CompareStrings(actualNames, expectedNames)
}

func ToStringSliceFunc[T any](elements []T, f func(T) string) []string {
	var result []string
	for _, e := range elements {
		result = append(result, f(e))
	}
	return result
}
```
so that we do not need to have the compare method for string, data source and resource slices seperately.